File: packages/contracts/src/ecosystem.ts
/**
 * The seed ecosystem the footer's Partners page shows (#79): partners with the verticals they
 * support, universities with their cohorts, and the licensable assets `reuse-fit` can pick.
 * Mirrors `services/engine/app/models/ecosystem.py` (EngineModel, so plain objects); the parity test keeps the two equal.
 */
import { z } from "zod";

import { Vertical } from "./brief.js";

export const EcosystemPartner = z.object({
  partnerId: z.string(),
  verticals: z.array(Vertical),
});
export type EcosystemPartner = z.infer<typeof EcosystemPartner>;

export const EcosystemUniversity = z.object({
  universityId: z.string(),
  cohorts: z.array(z.string()),
});
export type EcosystemUniversity = z.infer<typeof EcosystemUniversity>;

export const EcosystemAsset = z.object({
  assetId: z.string(),
  title: z.string(),
  vertical: Vertical,
});
export type EcosystemAsset = z.infer<typeof EcosystemAsset>;

export const Ecosystem = z.object({
  partners: z.array(EcosystemPartner),
  universities: z.array(EcosystemUniversity),
  assets: z.array(EcosystemAsset),
  /** Every seed entity is fictional (AGENTS.md rule 5). */
  demoData: z.boolean().default(true),
});
export type Ecosystem = z.infer<typeof Ecosystem>;

File: services/engine/app/models/ecosystem.py
"""The seed ecosystem as the footer's Partners page shows it (#79): partners with the verticals
they support, universities with their cohorts, and the licensable assets that `reuse-fit` can
pick. Read from graph predicates with plain `match` queries; no rule decides anything here.

Mirrors `packages/contracts/src/ecosystem.ts`; the parity test keeps the two equal.
"""

from app.models.brief import Vertical
from app.models.engine import EngineModel


class EcosystemPartner(EngineModel):
    partner_id: str
    verticals: list[Vertical]


class EcosystemUniversity(EngineModel):
    university_id: str
    cohorts: list[str]


class EcosystemAsset(EngineModel):
    asset_id: str
    title: str
    vertical: Vertical


class Ecosystem(EngineModel):
    partners: list[EcosystemPartner]
    universities: list[EcosystemUniversity]
    assets: list[EcosystemAsset]
    # Every seed entity is fictional (AGENTS.md rule 5).
    demo_data: bool = True
